### ISSUE TITLE
Getting Invalid URL remark-lint:double-link, false positive?

### ISSUE THREAD

**Issue by probonopd** [sindresorhus/awesome-lint#114]
Hello @sindresorhus I am getting a validation error which I suspect to be a false positive:

My list:

* Source: https://raw.githubusercontent.com/AppImage/awesome-appimage/e9dcc94fab9e43bb213083318a61a6964e6a0fc0/README.md
* Rendered: https://github.com/AppImage/awesome-appimage/blob/e9dcc94fab9e43bb213083318a61a6964e6a0fc0/README.md

The validation error:

```
1:1  Invalid URL: http:#appimage-discovery        remark-lint:double-link
```

Full log:

https://github.com/AppImage/awesome-appimage/runs/939913779?check_suite_focus=true#step:5:8

What does it mean, how can I fix it? Thank you very much.

**Comment by sindresorhus** [comment]
// @Scrum

**Comment by Scrum** [comment]
@probonopd @sindresorhus Thanks for the find, I have already made the necessary changes to the package `remark-lint-double-link`, let me know if something goes wrong after the update.

**Comment by probonopd** [comment]
Thank you very much @Scrum - this solves the particular issue.

But now I am getting what looks like other false positives:

```
  ✖  193:3   https://www.youtube.com/watch?v=XTGn_JqmDu0                                    remark-lint:double-link
  ✖  203:3   https://www.youtube.com/watch?v=nzZ6Ikc7juw                                    remark-lint:double-link
  ✖  204:3   https://www.youtube.com/watch?v=7fPShv-8Z_4                                    remark-lint:double-link
```

Could it be that what follows after the `?` is ignored?

**Comment by Scrum** [comment]
@probonopd Hi, it will take me a while to explore the possibility of including parameters in the plugin

**Comment by jthegedus** [comment]
@Scrum until then, is there an ignore for this particular rule but only applying to a specific URL? IE: ignore all YouTube links?

**Comment by Scrum** [comment]
@probonopd @jthegedus 

> @Scrum until then, is there an ignore for this particular rule but only applying to a specific URL? IE: ignore all YouTube links?

Yes, now you can set the ignore flag for such link types, for example

```
- [puper-link-1](http://link-1)
- [puper-link-1](http://link-2)
<!--lint ignore double-link-->
- [Produce an AppImage that bundles everything with go-appimage](https://www.youtube.com/watch?v=XTGn_JqmDu0)
- [puper-link-1](http://link-1)
- [AppImage: Portable applications for Linux](https://www.youtube.com/watch?v=nzZ6Ikc7juw)
- [Comparing Linux Package Formats - Deb, Flatpak, AppImage, etc.](https://www.youtube.com/watch?v=7fPShv-8Z_4)
```

In this case, you will receive only one error `'2:3-2:32: http://link-1',`

**Comment by ugurkanates** [comment]
Also happening to me as well.

**Comment by Scrum** [comment]
resolved in https://github.com/Scrum/remark-lint-double-link/pull/6 Now query parameters will participate in validation
